src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,105 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationServiceImpl registrationService;
+    private List<User> users;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        users = new ArrayList<>();
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("validPassword");
+        user.setAge(23);
+        users.add(user);
+    }
+
+    @Test
+    void userExistsAndLoginAndPasswordAreValid_Ok() {
+        User user = users.get(0);
+        User registeredUser = registrationService.register(user);
+        assertNotNull(registeredUser);
+        assertEquals(user, registeredUser);
+    }
+
+    @Test
+    void userDoesNotExist_NotOk() {
+        User user = new User();
+        assertFalse(Storage.people.contains(user));
+    }
+
+    @Test
+    void loginIsLessThanMinimalLength_NotOk() {
+        User user = users.get(0);
```

Comment:
The list is redundant, just use the user, that has been initialized in "before each"